olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -183,18 +183,28 @@ client::CancellationToken VersionedLayerClientImpl::PrefetchPartitions(
       return;
     }
 
+    const auto key = request.CreateKey(layer_id_);
+
+    if (!settings_.cache) {
+      OLP_SDK_LOG_ERROR_F(
+          kLogTag,
+          "PrefetchPartitions: cache is missing, aborting, hrn=%s, key=%s",
+          catalog_.ToCatalogHRNString().c_str(), key.c_str());
+      callback(ApiError(ErrorCode::PreconditionFailed,
```

Comment:
This `ErrorCode::PreconditionFailed` seems to repeat very often. I would propose to add a new static method to ApiError:
```cpp
class CORE_API ApiError {
    static ApiError PreconditionFailed(const char* description = "Precondition failed") {
        return ApiError(ErrorCode::PreconditionFailed, description);
  }
};
```

Reply:
Do you want me to fix each usage?

Reply:
Only for the files you touched, should be enough.